dev/integration_tests/release_smoke_test/ios/Podfile

```diff
@@ -15,49 +15,64 @@ def parse_KV_file(file, separator='=')
   if !File.exists? file_abs_path
     return [];
   end
-  pods_ary = []
+  generated_key_values = {}
```

Comment:
Copy of the template.

Reply:
Do these mean this will break for consumers?

Reply:
No, the original format is the same, and this method isn't accessible outside this file (unless someone was doing some really weird hackery).

Reply:
Sorry, I mean the fact that you had to update all of these - will it not be automatic for people running Flutter on existing projects?

Reply:
The only place this method is called is [within](https://github.com/flutter/flutter/pull/42029/files#diff-b4f1b78922d84823776457d645d50192R50) this [Podfile](https://github.com/flutter/flutter/pull/42029/files#diff-b4f1b78922d84823776457d645d50192R70).  So if they regenerate the Podfile they will get the updated method and the updated usage.

Reply:
Oh I understand your question.  Nothing will break for existing customers.  They just won't use the copied version, and instead will keep using the symlink and may keep experiencing the old stale framework issue.  If they regenerated their Podfile, they will get the new behavior.

Reply:
Could we have the tool check if they're going to be using the symlink and warn them that it could cause problems and either fix it or provide instructions?

Reply:
I don't think we can fix it automatically since that would involve overwriting their Podfile, and people seem to edit that and expect it to not be removed or overwritten.
Maybe something like https://github.com/flutter/flutter/blob/cdb3578b140b3708dc47a958c538b1501f2b6106/packages/flutter_tools/lib/src/resident_runner.dart#L970 ?

After `pod install` the tool could check for the presence of `.symlinks/flutter` and print a warning their Podfile is out of date and they should regenerate?
```
'Your Podfile is out of date.\nConsider removing ios/Podfile and running "flutter create ." to regenerate it.';
```
That might be annoying since it doesn't give people who already edited their Podfile a way to just fix up the problem.
@zanderso What do you think?

Reply:
P.S. The stale framework issue this is fixing happens when developers fail to follow the deployment instructions and do not run `flutter build ios` in release before archiving. So if the Podfile isn't updated, but they follow instructions, they will be fine.

Reply:
Maybe we could read the podfile an dlook for the the offending line?

Reply:
A more detailed message, or a link to some docs on the website, could help folks do the necessary update manually if they have local edits they want to keep. Like, "Your Podfile is out of date. Regenerate it with .... If you have local edits that you'd like to keep, see ... for instructions."

Reply:
It's easy to see if their Podfile is out of date without parsing (either `.symlinks/flutter` exists or it doesn't since that directory)
The minimum Podfile change (without the mapping refactor) is pretty large, so it's not just a one-line swap:
```ruby
generated_xcode_build_settings = parse_KV_file('./Flutter/Generated.xcconfig')
if generated_xcode_build_settings.empty?
  puts "Generated.xcconfig must exist. If you're running pod install manually, make sure flutter pub get is executed first."
end
generated_xcode_build_settings.map { |p|
  if p[:name] == 'FLUTTER_FRAMEWORK_DIR'
    symlink = File.join('.symlinks', 'flutter')
    File.symlink(File.dirname(p[:path]), symlink)
    pod 'Flutter', :path => File.join(symlink, File.basename(p[:path]))
  end
}
```
becomes:
```ruby
copied_flutter_dir = File.join(__dir__, 'Flutter')
copied_framework_path = File.join(copied_flutter_dir, 'Flutter.framework')
copied_podspec_path = File.join(copied_flutter_dir, 'Flutter.podspec')
unless File.exist?(copied_framework_path) && File.exist?(copied_podspec_path)
  generated_xcode_build_settings = parse_KV_file('./Flutter/Generated.xcconfig')
  if generated_xcode_build_settings.empty?
    puts "Generated.xcconfig must exist. If you're running pod install manually, make sure flutter pub get is executed first."
  end
  generated_xcode_build_settings = parse_KV_file(generated_xcode_build_settings_path)
  generated_xcode_build_settings.map { |p|
    if p[:name] == 'FLUTTER_FRAMEWORK_DIR'
      cached_framework_dir = p[:path]
      unless File.exist?(copied_framework_path)
        FileUtils.cp_r(File.join(cached_framework_dir, 'Flutter.framework'), copied_flutter_dir)
      end
      unless File.exist?(copied_podspec_path)
        FileUtils.cp(File.join(cached_framework_dir, 'Flutter.podspec'), copied_flutter_dir)
      end
    end
  }
end

pod 'Flutter', :path => 'Flutter'
```

Reply:
Here's what it looks like when I have an old version, build, get the migration instructions, remove the Podfile, then build again:
![podfile_migration](https://user-images.githubusercontent.com/682784/66356379-9683f880-e91f-11e9-8ab6-5285b74da8c9.png)